<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>EmulationSettingsWidget</class>
 <widget class="QWidget" name="EmulationSettingsWidget">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>618</width>
    <height>481</height>
   </rect>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <property name="leftMargin">
    <number>0</number>
   </property>
   <property name="topMargin">
    <number>0</number>
   </property>
   <property name="rightMargin">
    <number>0</number>
   </property>
   <property name="bottomMargin">
    <number>0</number>
   </property>
   <item>
    <widget class="QGroupBox" name="groupBox_5">
     <property name="title">
      <string>Speed Control</string>
     </property>
     <layout class="QGridLayout" name="gridLayout_2" columnstretch="0,1">
      <item row="0" column="0">
       <widget class="QLabel" name="label_4">
        <property name="text">
         <string>Emulation Speed:</string>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="QComboBox" name="emulationSpeed"/>
      </item>
      <item row="1" column="0">
       <widget class="QLabel" name="label_5">
        <property name="text">
         <string>Fast Forward Speed:</string>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <widget class="QComboBox" name="fastForwardSpeed"/>
      </item>
      <item row="2" column="0">
       <widget class="QLabel" name="label_6">
        <property name="text">
         <string>Turbo Speed:</string>
        </property>
       </widget>
      </item>
      <item row="2" column="1">
       <widget class="QComboBox" name="turboSpeed"/>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="groupBox_2">
     <property name="title">
      <string>Latency Control</string>
     </property>
     <layout class="QGridLayout" name="gridLayout">
      <item row="0" column="0">
       <widget class="QCheckBox" name="vsync">
        <property name="text">
         <string>Vertical Sync (VSync)</string>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="QCheckBox" name="syncToHostRefreshRate">
        <property name="text">
         <string>Sync To Host Refresh Rate</string>
        </property>
       </widget>
      </item>
      <item row="1" column="0">
       <widget class="QCheckBox" name="optimalFramePacing">
        <property name="text">
         <string>Optimal Frame Pacing</string>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <widget class="QCheckBox" name="preFrameSleep">
        <property name="text">
         <string>Reduce Input Latency</string>
        </property>
       </widget>
      </item>
      <item row="2" column="0">
       <widget class="QCheckBox" name="skipPresentingDuplicateFrames">
        <property name="text">
         <string>Skip Duplicate Frame Display</string>
        </property>
       </widget>
      </item>
      <item row="2" column="1">
       <layout class="QFormLayout" name="formLayout_3">
        <item row="0" column="0">
         <widget class="QLabel" name="preFrameSleepBufferLabel">
          <property name="text">
           <string>Frame Time Buffer:</string>
          </property>
         </widget>
        </item>
        <item row="0" column="1">
         <widget class="QDoubleSpinBox" name="preFrameSleepBuffer">
          <property name="suffix">
           <string> Milliseconds</string>
          </property>
          <property name="decimals">
           <number>1</number>
          </property>
          <property name="singleStep">
           <double>0.500000000000000</double>
          </property>
         </widget>
        </item>
       </layout>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="groupBox">
     <property name="title">
      <string>Rewind</string>
     </property>
     <layout class="QGridLayout" name="gridLayout_3" columnstretch="0,0">
      <item row="1" column="0">
       <widget class="QLabel" name="label_2">
        <property name="text">
         <string>Rewind Save Frequency:</string>
        </property>
       </widget>
      </item>
      <item row="2" column="1">
       <widget class="QSpinBox" name="rewindSaveSlots">
        <property name="suffix">
         <string> Frames</string>
        </property>
        <property name="minimum">
         <number>1</number>
        </property>
        <property name="maximum">
         <number>10000</number>
        </property>
       </widget>
      </item>
      <item row="3" column="0" colspan="2">
       <widget class="QLabel" name="rewindSummary">
        <property name="text">
         <string>TextLabel</string>
        </property>
        <property name="wordWrap">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item row="2" column="0">
       <widget class="QLabel" name="label">
        <property name="text">
         <string>Rewind Buffer Size:</string>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <widget class="QDoubleSpinBox" name="rewindSaveFrequency">
        <property name="suffix">
         <string> Seconds</string>
        </property>
        <property name="maximum">
         <double>3600.000000000000000</double>
        </property>
        <property name="singleStep">
         <double>0.100000000000000</double>
        </property>
       </widget>
      </item>
      <item row="0" column="0">
       <widget class="QCheckBox" name="rewindEnable">
        <property name="text">
         <string>Enable Rewinding</string>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="QCheckBox" name="rewindUseSaveStates">
        <property name="text">
         <string>Rewind using Save States</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="groupBox_3">
     <property name="title">
      <string>Runahead</string>
     </property>
     <layout class="QGridLayout" name="gridLayout_4">
      <item row="0" column="0">
       <widget class="QLabel" name="label_3">
        <property name="text">
         <string>Runahead:</string>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="QComboBox" name="runaheadFrames">
        <item>
         <property name="text">
          <string>Disabled</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>1 Frame</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>2 Frames</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>3 Frames</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>4 Frames</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>5 Frames</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>6 Frames</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>7 Frames</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>8 Frames</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>9 Frames</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>10 Frames</string>
         </property>
        </item>
       </widget>
      </item>
      <item row="1" column="0" colspan="2">
       <widget class="QCheckBox" name="runaheadForAnalogInput">
        <property name="text">
         <string>Enable for Analog Input</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <spacer name="verticalSpacer">
     <property name="orientation">
      <enum>Qt::Orientation::Vertical</enum>
     </property>
     <property name="sizeHint" stdset="0">
      <size>
       <width>0</width>
       <height>0</height>
      </size>
     </property>
    </spacer>
   </item>
  </layout>
 </widget>
 <resources>
  <include location="resources/duckstation-qt.qrc"/>
 </resources>
 <connections/>
</ui>
